### PR TITLE
[Fix]: [CDS-39434]: added coverage for some jira files

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35217)

### CODE REVIEW

File: 800-pipeline-service/src/test/java/io/harness/pms/approval/ApprovalinstanceHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+public class ApprovalinstanceHandlerTest extends CategoryTest {
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+
+  @Mock private PipelineServiceIteratorsConfig iteratorsConfig;
+  @Mock private JiraApprovalHelperService jiraApprovalHelperService;
+  @Mock private MongoTemplate mongoTemplate;
+  @Mock private PersistenceIteratorFactory persistenceIteratorFactory;
+  @Mock private ServiceNowApprovalHelperService serviceNowApprovalHelperService;
+  @Mock private ApprovalInstance approvalInstance;
```
Comment (puthrayaharness) [inline 800-pipeline-service/src/test/java/io/harness/pms/approval/ApprovalinstanceHandlerTest.java:44]: Same do not mock

File: 800-pipeline-service/src/test/java/io/harness/pms/approval/ApprovalinstanceHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  }
+  @Test
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void testhandleservice() {
+    ApprovalInstanceHandler approvalInstanceHandler = new ApprovalInstanceHandler(jiraApprovalHelperService,
+        mongoTemplate, persistenceIteratorFactory, iteratorsConfig, serviceNowApprovalHelperService);
+    ApprovalInstance entity = (ApprovalInstance) serviceNowApprovalInstance;
+    when(entity.getType()).thenReturn(SERVICENOW_APPROVAL);
+    approvalInstanceHandler.handle(entity);
```
Comment (puthrayaharness) [inline 800-pipeline-service/src/test/java/io/harness/pms/approval/ApprovalinstanceHandlerTest.java:85]: Same

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java [hunk 3]
```diff
@@ -80,6 +98,86 @@ public void testValidateCredentialsExactError() throws Exception {
...
+        .isInstanceOf(JiraTaskNGResponse.class);
+  }
+
+  @Test
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void testgetStatus() throws Exception {
+    JiraClient jiraClient = Mockito.mock(JiraClient.class);
+    when(jiraClient.getStatuses(anyString(), anyString())).thenReturn(Collections.emptyList());
+    PowerMockito.whenNew(JiraClient.class).withAnyArguments().thenReturn(jiraClient);
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java:119]: Please remove usage of PowerMockito.

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java [hunk 4]
```diff
@@ -80,6 +98,86 @@ public void testValidateCredentialsExactError() throws Exception {
             "Check if the Jira URL & Jira credentials are correct. Jira URLs are different for different credentials");
   }
 
+  @Test
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void testgetprojects() throws Exception {
+    JiraClient jiraClient = Mockito.mock(JiraClient.class);
+    when(jiraClient.getProjects()).thenReturn(Collections.emptyList());
+    PowerMockito.whenNew(JiraClient.class).withAnyArguments().thenReturn(jiraClient);
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java:107]: Please remove usage of PowerMockito

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java [hunk 5]
```diff
@@ -80,6 +98,86 @@ public void testValidateCredentialsExactError() throws Exception {
...
+  }
+
+  @Test
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void testgetIssue() throws Exception {
+    JiraClient jiraClient = Mockito.mock(JiraClient.class);
+    JiraIssueNG jiraIssueNG = new JiraIssueNG();
+    when(jiraClient.getIssue(anyString())).thenReturn(jiraIssueNG);
+    PowerMockito.whenNew(JiraClient.class).withAnyArguments().thenReturn(jiraClient);
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java:132]: Please remove usage of Powermockito

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java [hunk 6]
```diff
@@ -80,6 +98,86 @@ public void testValidateCredentialsExactError() throws Exception {
...
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void testcreateIssue() throws Exception {
+    JiraClient jiraClient = Mockito.mock(JiraClient.class);
+
+    JiraIssueNG jiraIssueNG = new JiraIssueNG();
+    jiraIssueNG.setId("123");
+    jiraIssueNG.setKey("abc");
+    when(jiraClient.createIssue(anyString(), anyString(), anyMap(), anyBoolean())).thenReturn(jiraIssueNG);
+    PowerMockito.whenNew(JiraClient.class).withAnyArguments().thenReturn(jiraClient);
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java:148]: Please remove usage of Powermockito

File: 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java [hunk 7]
```diff
@@ -80,6 +98,86 @@ public void testValidateCredentialsExactError() throws Exception {
...
+  @Owner(developers = vivekveman)
+  @Category(UnitTests.class)
+  public void testupdateIssue() throws Exception {
+    JiraClient jiraClient = Mockito.mock(JiraClient.class);
+
+    JiraIssueNG jiraIssueNG = new JiraIssueNG();
+    jiraIssueNG.setId("123");
+    jiraIssueNG.setKey("abc");
+    when(jiraClient.updateIssue(anyString(), anyString(), anyString(), anyMap())).thenReturn(jiraIssueNG);
+    PowerMockito.whenNew(JiraClient.class).withAnyArguments().thenReturn(jiraClient);
```
Comment (puthrayaharness) [inline 930-delegate-tasks/src/test/java/io/harness/delegate/task/jira/JiraTaskNGHandlerTest.java:169]: Please remove usage of PowerMockito

File: 800-pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/ApprovalInstanceServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+import org.springframework.transaction.support.TransactionTemplate;
+
+@OwnedBy(CDC)
+public class ApprovalInstanceServiceTest extends CategoryTest {
+  @Rule public MockitoRule mockitoRule = MockitoJUnit.rule();
+  @Mock private ApprovalInstanceRepository approvalInstanceRepository;
+  @Mock private TransactionTemplate transactionTemplate;
+  @Mock private WaitNotifyEngine waitNotifyEngine;
+  @Mock private PlanExecutionService planExecutionService;
+  @Mock private ApprovalInstance approvalInstance;
```
Comment (puthrayaharness) [inline 800-pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/ApprovalInstanceServiceTest.java:50]: Remove mock

File: 800-pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/ApprovalInstanceServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+  @Mock private UpdateResult updateResult;
```
Comment (puthrayaharness) [inline 800-pipeline-service/modules/orchestration-steps/src/test/java/io/harness/steps/approval/ApprovalInstanceServiceTest.java:51]: Remove global mock and move this to method. Use `UpdateResult updateResult = mock(UpdateResult.class)`

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger ti1

**Comment by vivekkarnatiharness** [comment]
trigger ti1

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35217)* status: :shipit: complete! ✓

**Review by gaurav-jain1 — APPROVED** [review]
> Reviewed 1 of 2 files at r1, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35217)* status: :shipit: complete! ✓

**Pull request merged**